src/components/organisms/CesiumViewer.vue

```diff
@@ -3,44 +3,115 @@ import {
   defineComponent,
   onMounted,
   ref,
-  toRef,
   watch,
 }
   from '@vue/composition-api';
 import Cesium from '@/plugins/cesium';
+import { useMap, geoShape } from '@/store';
 
 export default defineComponent({
   name: 'CesiumViewer',
-  props: {
-    location: {
-      type: Object, /* TODO typescript interfaces for props */
-      required: false,
-      default: () => ({}),
-    },
-  },
-  setup(props) {
+  setup() {
+    const polyPoints: any[] = [];
     const cesiumViewer = ref();
     onMounted(() => {
       cesiumViewer.value = new Cesium.Viewer('cesiumContainer', {
         animation: false,
         timeline: false,
+        fullscreenButton: false,
+        infoBox: false,
+        selectionIndicator: false,
+        terrainProvider: Cesium.createWorldTerrain(),
       });
       cesiumViewer.value.forceResize();
       cesiumViewer.value.camera.flyTo({
         destination: Cesium.Cartesian3.fromDegrees(-93.849688, 40.690265, 4000000),
       });
     });
-    const updateCamera = () => {
-      cesiumViewer.value.camera.flyTo({
-        destination: Cesium.Cartesian3.fromDegrees(
-          props.location.longitude,
-          props.location.latitude,
-          props.location.height,
-        ),
-      });
-    };
-    watch(toRef(props, 'location'), updateCamera, {
-      deep: true,
+    watch(useMap, (val) => {
+      if (!val) { return; }
+      {
+        cesiumViewer.value.cesiumWidget.screenSpaceEventHandler.removeInputAction(
+          Cesium.ScreenSpaceEventType.LEFT_DOUBLE_CLICK,
+        );
+        const createPoint = (worldPosition: any) => {
```

Comment:
FWIW, Cesium apparently does actually have type definitions (https://cesium.com/blog/2020/06/01/cesiumjs-tsd/), and importing them appears to work without any issues with our current webpack setup. I think making use of them would add value and make the CesiumJS-related code more maintainable.

Just something I wanted to bring up for the future, not something to worry about for this PR, and probably not before the first iteration of the web app if it's anything more then adding some type annotations.

Reply:
Good to know!